### PR TITLE
[solo.commands.mark] new command

### PR DESCRIPTION
**Pull request by Enlik**
Introduce new command that can mark different properties on installed packages;
currently auto/manual. Command syntax for this resembles the one of apt-mark.                              

Example:

    # equo q list installed --by-user | grep gitweb
    # equo mark manual gitweb
    ╠  !! www-apps/gitweb-2.18.0: dependency => manual
    # equo q list installed --by-user | grep gitweb
    ╠ # 26021 www-apps/gitweb
    # equo mark auto gitweb
    ╠  !! www-apps/gitweb-2.18.0: manual => dependency
    # equo q list installed --by-user | grep gitweb
    #

Uses the newly introduced setInstalledPackageSource() in equo.db.

### CODE REVIEW

**Comment by Enlik** [comment]
I made it on top of changes for PR #49 so only the commit unique to this PR is relevant. It's better to take care of #49 first (reviews can be done in parallel, just ignore duplicated commits).

**Comment by Enlik** [comment]
The functionality overall (i.e. from both PRs) is nice because it should help with the scenario:

- on update, new dependencies: a, b and c are pulled,
- on another update, b and c is no longer needed,
- user could be able to drop them.

The problem is that current systems don't have proper auto/manual status on most packages, so it's ok for: new packages, installed after these fixes (hopefully! it should be much better at least; we'll see); existing systems if user uses `apt mark` (which is completely optional); after some work, on new installations.

File: client/doc/mansrc/equo-mark.1.txt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    mark package as installed to satisfy a dependency
+
+*manual*::
+    mark package as installed by user
+
+
+
+AUTHORS
+-------
+Fabio Erculiani (lxnay@sabayon.org)
```
Comment (mudler) [inline client/doc/mansrc/equo-mark.1.txt]: Well, i wonder why not adding yourself here? :P you are author of this command at end

File: lib/entropy/spm/plugins/interfaces/portage_plugin/__init__.py [hunk 2]
```diff
@@ -3468,6 +3458,38 @@ def __remove_update_world_file(self, key, slot):
...
+        except (OSError, IOError) as err:
+            if err.errno != errno.ENOENT:
+                raise
+
+        if keyslot not in world_atoms and \
+            entropy.tools.istextfile(world_file):
+
+            world_atoms.discard(key)
+            world_atoms.add(keyslot)
+            world_file_tmp = world_file+".entropy_inst"
```
Comment (mudler) [inline lib/entropy/spm/plugins/interfaces/portage_plugin/__init__.py]: :+1:

File: client/doc/mansrc/equo-mark.1.txt (hunk 1 above)
Comment (Enlik) [reply client/doc/mansrc/equo-mark.1.txt]: This file is auto-generated (client/doc/generate).

I can update `generate` to pull author from docstring if you think it'll be useful (especially for future contributions). Maybe in another PR when I'd also update docs? :)

File: lib/entropy/spm/plugins/interfaces/portage_plugin/__init__.py (hunk 2 above)
Comment (mudler) [reply lib/entropy/spm/plugins/interfaces/portage_plugin/__init__.py]: Appreciate the re-order :)

**Pull request merged**